File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        countMap = {}
        for num in nums:
            countMap[num] = 1 + countMap.get(num, 0)
        
        arr = []
        for num, count in countMap.items():
            arr.append([count,num])
        arr.sort()

        res = []
        while len(res) < k:
            res.append(arr.pop()[1])
        return res

File: Data Structures & Algorithms/subsets-ii/submission-0.cpp
class Solution {
private:
    vector<vector<int>> res;
    void backtrack(vector<int> &nums, vector<int> &subset, int start) {
        res.push_back(subset);
        for(int i = start; i < nums.size(); i++) {
            if ( i > start and nums[i] == nums[i-1]) continue;
            subset.push_back(nums[i]);
            backtrack(nums, subset, i + 1);
            subset.pop_back();
        }
    }
public:
    vector<vector<int>> subsetsWithDup(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        vector<int> subset;
        backtrack(nums, subset, 0);
        return res;
    }
};
